code/media/src/main/java/com/adobe/marketing/mobile/media/internal/MediaSession.java

```diff
@@ -272,15 +274,24 @@ private void trySendHit() {
                             Log.debug(
                                     MediaInternalConstants.EXTENSION_LOG_TAG,
                                     LOG_TAG,
-                                    "trySendHit - (%s) Finished http connection",
+                                    "trySendHit - (%s) Finished http connection.",
                                     hitEventType);
 
                             synchronized (mutex) {
                                 boolean shouldRetry = false;
 
-                                if (hitIsSessionStart
+                                if (deviceOffline) {
```

Comment:
can we use simply connection == null here

Reply:
I prefer having a boolean as makes it easier to read the code.